infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/configuration/impl/InfinispanRemoteConfiguration.java

```diff
@@ -87,4 +106,36 @@ public void initConfiguration(Map<?, ?> configurationMap, ServiceRegistryImpleme
 
 		log.tracef( "Initializing Infinispan Hot Rod client from configuration file at '%1$s'", configurationResource );
 	}
+
+	/**
+	 * Extract from the configuration the Hot Rod client properties: the one prefixed with
+	 * {@link InfinispanRemoteProperties#HOT_ROD_CLIENT_PREFIX} or defined in the resource file via
+	 * {@link InfinispanRemoteProperties#CONFIGURATION_RESOURCE_NAME}.
+	 * <p>
+	 * Note that the properties with the prefix will override the same properties in the resource file.
+	 */
+	private Properties getHotRodConfiguration(Map<?, ?> configurationMap, ConfigurationPropertyReader propertyReader, URL configurationResourceUrl) {
+		Properties hotRodConfiguration = new Properties();
+		if ( configurationResourceUrl != null ) {
+			try ( InputStream openStream = configurationResourceUrl.openStream() ) {
+				hotRodConfiguration.load( openStream );
+			}
+			catch (IOException e) {
+				throw log.failedLoadingHotRodConfigurationProperties( e );
+			}
+		}
+		// Programmatic properties override the resource file
+		for ( Entry<?, ?> property : configurationMap.entrySet() ) {
+			String key = (String) property.getKey();
+			if ( key.startsWith( HOT_ROD_CLIENT_PREFIX ) ) {
+				String value = propertyReader.property( key, String.class ).withDefault( null ).getValue();
```

Comment:
.withDefault( null )

This is unnecessary right? A not found key would mean something is wrong with our configuration parsing code; maybe let it throw an exception instead?